packages/block-library/src/navigation-link/link-ui.js

```diff
@@ -186,54 +201,145 @@ export function LinkUI( props ) {
 		[ label, opensInNewTab, url ]
 	);
 
+	const dialogTitleId = useInstanceId(
+		LinkUI,
+		`link-ui-link-control__title`
+	);
+	const dialogDescritionId = useInstanceId(
+		LinkUI,
+		`link-ui-link-control__description`
+	);
+
+	// Selecting a block should close the popover and also remove the (previously) automatically inserted
+	// link block so that the newly selected block can be inserted in its place.
+	const onSelectBlock = props.onClose;
+
 	return (
-		<Popover
-			placement="bottom"
-			onClose={ props.onClose }
-			anchor={ props.anchor }
-			shift
-		>
-			<LinkControl
-				hasTextControl
-				hasRichPreviews
-				value={ link }
-				showInitialSuggestions={ true }
-				withCreateSuggestion={ userCanCreate }
-				createSuggestion={ handleCreate }
-				createSuggestionButtonText={ ( searchTerm ) => {
-					let format;
-
-					if ( type === 'post' ) {
-						/* translators: %s: search term. */
-						format = __( 'Create draft post: <mark>%s</mark>' );
-					} else {
-						/* translators: %s: search term. */
-						format = __( 'Create draft page: <mark>%s</mark>' );
-					}
-
-					return createInterpolateElement(
-						sprintf( format, searchTerm ),
-						{
-							mark: <mark />,
-						}
-					);
-				} }
-				noDirectEntry={ !! type }
-				noURLSuggestion={ !! type }
-				suggestionsQuery={ getSuggestionsQuery( type, kind ) }
-				onChange={ props.onChange }
-				onRemove={ props.onRemove }
-				onCancel={ props.onCancel }
-				renderControlBottom={
-					! url
-						? () => (
-								<LinkControlTransforms
-									clientId={ props.clientId }
-								/>
-						  )
-						: null
-				}
-			/>
-		</Popover>
+		<>
+			{ showBackdrop && (
+				<div
+					className="components-popover-pointer-events-trap"
+					aria-hidden="true"
```

Comment:
I'm unsure. I copied the pattern from another part of the codebase. 

> aria-hidden="true" should not be used on elements that can receive focus.

[MDN suggests](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden) this isn't required.

Reply:
`aria-hidden` is used to hide things from the accessibility tree, and if there's no content to hide, then, AFAIK, it shouldn't do anything. Also, I think we shouldn't do this popover hack if at all possible. I think we should try to solve the underlying issue.

Reply:
> Also, I think we shouldn't do this popover hack if at all possible. I think we should try to solve the underlying issue.

I agree that we should resolve the underlying issue, however it's looking like a major issue that's affecting multiple parts of the codebase as is unrelated to the changes here.

@ciampo has used a similar hotfix as a temporary measure to work around the same problem in another PR.

Are we sure we want to block what it is a very useful feature because of this?

Reply:
I agree, let's merge and capture this in a followup issue.

Reply:
Created a followup here: https://github.com/WordPress/gutenberg/issues/58825